src/base/IoManip.h

```diff
@@ -76,27 +76,31 @@ operator <<(std::ostream &os, const RawPointerT<Pointer> &pd)
 }
 
 /// std::ostream manipulator to print integers as hex numbers prefixed by 0x
+/// Normally used through the asHex() convenience function
 template <class Integer>
 class AsHex
 {
 public:
-    explicit AsHex(const Integer n): io_manip(n) {}
+    explicit AsHex(const Integer n, const size_t w=0) : io_manip(n), width(w) {}
```

Comment:
Please use a named method to configure output, not an (anonymous from the caller point of view) constructor parameter. See class Raw for examples. Among other benefits, this approach does not result in duplication of new construction parameters in the asHex() wrapper.

The functionality this PR is currently adding goes way beyond specifying the width (which could probably be done using an existing std::setw() manipulator _in the asHex() caller_). Thus, the new method name should not be width() or similar. I can suggest `withLeadingZeros(size_t minDigits)`.

Reply:
Unfortunately setw() won't work (I tried), with std::hex, because the output would then look like '0000x0abcd'. This is also the reason why I have changed how the prefix is added